flink-table-store-core/src/main/java/org/apache/flink/table/store/file/casting/AbstractToStringCastExecutor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.casting;
+
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.binary.BinaryStringData;
+import org.apache.flink.table.data.binary.BinaryStringDataUtil;
+
+import javax.annotation.Nullable;
+
+/**
+ * Cast input to string base implementation.
+ *
+ * @param <IN> the input value type.
+ */
+public abstract class AbstractToStringCastExecutor<IN> implements CastExecutor<IN, StringData> {
+    protected final boolean targetCharType;
+    protected final int targetLength;
+
+    AbstractToStringCastExecutor(boolean targetCharType, int targetLength) {
+        this.targetCharType = targetCharType;
+        this.targetLength = targetLength;
+    }
+
+    @Nullable
+    @Override
+    public StringData cast(@Nullable IN value) throws TableException {
+        boolean inputIsNull = value == null;
```

Comment:
if (is null) return null;
You may have misunderstood the codegen in Flink.
Please add tests too.

Reply:
I found it should return null for string to string, but for other value type such as numeric to string, it will return "" for null input in codegen, is it right? I have added `StringData getNullOutput()` to do with it

Reply:
The result is:
```
isNull = true;
value = "";
```
It is a null.

Reply:
As `LogicalTypeCast` doesn't support converting other value type to string, I think we can support StringToString here and return null directly.